src/main/kotlin/no/nav/familie/tilbake/kravgrunnlag/domain/Kravgrunnlag431.kt

```diff
@@ -57,7 +58,21 @@ data class Kravgrunnlagsperiode432(@Id
                                    @Version
                                    val versjon: Long = 0,
                                    @Embedded(onEmpty = Embedded.OnEmpty.USE_EMPTY)
-                                   val sporbar: Sporbar = Sporbar())
+                                   val sporbar: Sporbar = Sporbar()) {
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || javaClass != other.javaClass) return false
+        val that = other as Kravgrunnlagsperiode432
+        return this.månedligSkattebeløp == that.månedligSkattebeløp &&
+               this.periode == that.periode &&
+               this.beløp == that.beløp
+    }
+
+    override fun hashCode(): Int {
+        return Objects.hash(id, periode, månedligSkattebeløp, beløp)
+    }
```

Comment:
Her burde du ikke override equals. Med en så tynn equals-funksjon, så kan andre utviklere få stygge overaskelser i femtiden. Det hadde vært bedre å lage en ny metode med et funksjonelt navn "harLiktPeriodegrunnlagSom" eller noe lignedne.

Reply:
@bent-lorentzen denne equals metoden inneholder alle felter til Kravgrunnlagsperiode432 bortsett fra id felt.

Reply:
Der så jeg feil, måptte ekspandere litt kode, men equals, hascode-kontrakten sier at hvis to objekter er like, så må hashkoden være lik, så noe må endre her.

Reply:
Synes du fortsatt at vi trenger et separat metode?

Reply:
Ikke hvis du kan få en gyldig hashkode for like objekter.

Reply:
fikset, endret metodenavn til harIdentiskKravgrunnlagsperiode